tests/pyminisketch.py

```diff
@@ -0,0 +1,514 @@
+                for _ in range(field_size):
+                    xp = gf.sqr(xp)
+                self.assertEqual(xp, x) # x^(2^field_size) == x
+            if y != 0:
+                yi = gf.inv(y)
+                self.assertEqual(y == yi, y == 1) # y==1/x iff y==1
+                self.assertEqual(gf.mul(y, yi), 1) # y*(1/y) == 1
+                yii = gf.inv(yi)
+                self.assertEqual(y, yii) # 1/(1/y) == y
+                if x != 0:
+                    xi = gf.inv(x)
+                    xyi = gf.inv(xy)
+                    self.assertEqual(xyi, gf.mul(xi, yi)) # (1/x)*(1/y) == 1/(x*y)
+
+    def test(self):
+        """Run tests."""
+        for field_size in range(2, 65):
+            self.field_size_test(field_size)
+
+# The operations below operate on polynomials over GF(2^field_size), represented as lists of
+# integers:
+#
+#   [a, b, c, ...] = a + b*x + c*x^2 + ...
+#
+# As an invariant, there are never any trailing zeroes in the list representation.
+#
+# Examples:
+# * [] = 0
+# * [3] = 3
+# * [0, 1] = x
+# * [2, 0, 5] = 5*x^2 + 2
+
+def poly_monic(poly, gf):
+    """Return a monic version of the polynomial poly."""
+    # Multiply every coefficient with the inverse of the top coefficient.
+    inv = gf.inv(poly[-1])
+    return [gf.mul(inv, v) for v in poly]
+
+def poly_divmod(poly, mod, gf):
+    """Return the polynomial (quotient, remainder) of poly divided by mod."""
+    assert len(mod) > 0 and mod[-1] == 1 # Require monic mod.
+    if len(poly) < len(mod):
+        return ([], poly)
+    val = list(poly)
+    div = [0 for _ in range(len(val) - len(mod) + 1)]
+    while len(val) >= len(mod):
+        term = val[-1]
+        div[len(val) - len(mod)] = term
+        # If the highest coefficient in val is nonzero, subtract a multiple of mod from it.
+        val.pop()
+        if term != 0:
+            for x in range(len(mod) - 1):
+                val[1 + x - len(mod)] ^= gf.mul(term, mod[x])
+    # Prune trailing zero coefficients.
+    while len(val) > 0 and val[-1] == 0:
+        val.pop()
+    return div, val
+
+def poly_gcd(a, b, gf):
+    """Return the polynomial GCD of a and b."""
+    if len(a) < len(b):
+        a, b = b, a
+    # Use Euclid's algorithm to find the GCD of a and b.
+    # see https://en.wikipedia.org/wiki/Polynomial_greatest_common_divisor#Euclid's_algorithm.
+    while len(b) > 0:
+        b = poly_monic(b, gf)
+        (_, b), a = poly_divmod(a, b, gf), b
+    return a
+
+def poly_sqr(poly, gf):
+    """Return the square of polynomial poly."""
+    if len(poly) == 0:
+        return []
+    # In characteristic-2 fields, thanks to Frobenius' endomorphism ((a + b)^2 = a^2 + b^2),
+    # squaring a polynomial is easy: square all the coefficients and interleave with zeroes.
+    # E.g., (3 + 5*x + 17*x^2)^2 = 3^3 + (5*x)^2 + (17*x^2)^2.
+    # See https://en.wikipedia.org/wiki/Frobenius_endomorphism.
+    return [0 if i & 1 else gf.sqr(poly[i // 2]) for i in range(2 * len(poly) - 1)]
```

Comment:
note for future reviewer if helpful, I'm not that great at single line list comprehension olympics 😅  so I tested by comparing
```py
    squared = []
    for coefficient in poly:
        squared.append(gf.sqr(coefficient))
        if len(squared) >= 2 * len(poly) - 1:
            break
        squared.append(0)
    assert len(squared) == (len(poly) - 1) * 2 + 1 # n-degree poly -> 2n-degree when squared
    assert squared == [0 if i & 1 else gf.sqr(poly[i // 2]) for i in range(2 * len(poly) - 1)]
```